### PR TITLE
Clabot whitelist email domain

### PR DESCRIPTION
**Pull request by maoo**
Adds the ability to define `claEmailWhitelist` boolean at org level, to enable email domain whitelisting for the FINOS CLA bot, see https://github.com/finos/cla-bot/pull/161

### CODE REVIEW

File: src/metadata_tool/tools/generators.clj [hunk 1]
```diff
@@ -24,7 +24,12 @@
 (defn gen-clabot-whitelist
   []
   (println (tem/render "clabot-whitelist.ftl"
-                       {:github-ids (sort (mapcat :github-logins (md/people-with-clas)))})))
+                       {:github-ids (sort (mapcat :github-logins (md/people-with-clas)))
+                        :email-domains 
+                          (sort (flatten 
+                            (mapcat :domains 
+                              (filter #(:cla-email-whitelist %)
```
Comment (tschady) [inline src/metadata_tool/tools/generators.clj]: this is a bit better as `(filter :cla-email-whitelist (md/organizations-metadata))`
I'd also consider using `->>`, looks simpler to me but YMMV.  

I don't know what the data structure looks like anymore (I don't have access to the `metadata` repo), so it's hard to figure out, but `flatten` is usually a smell.  `filter`, then `mapcat`, then `flatten` might have a better solution.    Minor syntax improvement, so just a suggestion.

**Comment by tschady** [comment]
LGTM, but I cannot test locally, so hard to know for sure.

File: src/metadata_tool/tools/generators.clj (hunk 1 above)
Comment (maoo) [reply src/metadata_tool/tools/generators.clj]: TY @tschady !

Re `->>`, I'm trying to use it, but takes a bit of practice I guess, so I'll schedule it for a future code refactoring.

Re. the `flatten`, this is the JSON of the FINOS org in our metadata DB:

```
{
  "metadataVersion"  : "2.1.0",
  "organizationName" : "FINOS",
  "hasCCLA"          : true,
  "domains"          : [ "finos.org" ]
}
```

The flatten is supposed to "extract" the list of `domains` values. Happy to apply a better approach of course!

Comment (tschady) [reply src/metadata_tool/tools/generators.clj]: in that case you don't need `flatten` at all, it's not doing anything.  `mapcat` gets you what you need.

You use vscode, right?  Checkout Calva, I hear good things.  Can instantly swap between threaded and unrolled forms.

Comment (maoo) [reply src/metadata_tool/tools/generators.clj]: Wow, Calva is fantastic, thank you! Colour-coded parenthesis are super helpful.

Comment (pmonks) [reply src/metadata_tool/tools/generators.clj]: Note that so-called "Rainbow Parentheses" have a long tradition in the Lisp/Clojure community, and are available in most editors.

**Pull request merged**